v_4_9_0/worker_template.go

```diff
@@ -138,15 +138,32 @@ systemd:
       ExecStopPost=-/usr/bin/docker rm -f $NAME
       [Install]
       WantedBy=multi-user.target
+  - name: k8s-setup-download-hyperkube.service
+    enabled: true
+    contents: |
+      [Unit]
+      Description=Pulls hyperkube binary from image to local FS
+      After=docker.service
+      Requires=docker.service
+      [Service]
+      Type=oneshot
+      RemainAfterExit=yes
+      TimeoutStartSec=0
+      Environment="IMAGE={{ .RegistryDomain }}/{{ .Images.Kubernetes }}"
+      Environment="NAME=%p.service"
+      ExecStart=/bin/bash -c "/usr/bin/docker create --name $NAME $IMAGE && \
+        /usr/bin/docker cp $NAME:/hyperkube /opt/bin/hyperkube && \
+        /usr/bin/docker rm $NAME"
```

Comment:
I'd write this in a systemd way by leveraging more the features that come out of the box with systemd.

Something like

```
ExecPreStart=/usr/bin/docker create --name $NAME $IMAGE
ExecStart=/usr/bin/docker cp $NAME:/hyperkube /opt/bin/hyperkube
ExecStop=/usr/bin/docker rm $NAME"
```

More elegant I'd say

WDYT?

Reply:
Good idea. I wasn't sure how that would interact with a oneshot unit, but it does look nicer. I'll give it a try.